### PR TITLE
Add initial take at custom apps development page

### PR DESCRIPTION
**Pull request by theRealPadster**
Put this together. I think it's got the most basic info needed. The longer code block in the middle can probably be trimmed down; I copied chunks of code from Marketplace. Let me know if you have suggestions or questions that could be addressed.

### CODE REVIEW

File: docs/development/custom-apps.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
+---
+title: Custom Apps
+sidebar_position: 4
+---
+
+- [How to start](#how-to-start)
+- [Manifest file](#manifest-file)
+- [Common questions](#common-questions)
```
[the remark this answers is not in the document]
Comment (theRealPadster) [reply docs/development/custom-apps.md:8]: Ah, I was just following the format as on the [Development/Customization](https://spicetify.app/docs/development/customization) page. I can remove if you'd like.

[the remark this answers is not in the document]
Comment (afonsojramos) [reply docs/development/custom-apps.md]: Whoops, my bad, will remove from that

File: docs/development/custom-apps.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    state: {
+        data: {
+            title: "My sub page title",
+            content: "My sub page content",
+        },
+    },
+});
+```
+
+In order to render a different page it is loaded, you can check the `pathname` of the current page. In the main render method in your `index.js`, you can check the `pathname`, and render a different page component for different paths. The main path for your custom app will be the name of the folder (what goes in your [`config-xpui.ini`](customization#configs)).
```
Comment (afonsojramos) [inline docs/development/custom-apps.md:115]: In order to render a different page ~~it is loaded~~, you can check the \`pathname\` of the current page **within** \`index.js\`'s main render method, ~~you can check the \`pathname\`,~~ and render a different page component for different paths. The main path for your custom app will be the name of the folder (**which is the same that needs to be used in the \[\`config-xpui.ini\`\](customization#configs) configuration file**).

**Review by afonsojramos — CHANGES_REQUESTED** [review]

File: docs/development/custom-apps.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+- The `index.js` file is the main file for the custom app. 
+- The manifest includes some important information to make the custom app work. 
+
+## Manifest file
+- Your custom app needs a `manifest.json` file in the root folder with the following keys:
+- `name`: The name of the custom app. 
+- `icon`: The escaped SVG markup for the sidebar icon. 
+- `active-icon`: The escaped SVG markup for the active status of the sidebar icon (when your custom app is open). 
+- `subfiles`: You can optionally include other JS files. These files will be concatenated together in the order defined here. Any variables you delcare in the main `index.js`, or any subfiles will be accessible from all. This is useful for organizational purposes for more complex custom apps. 
+- `subfiles_extension`: You can optionally include an extension(s) with your custom app. These are treated as regular extensions, and will run when Spotify starts.
```
[the remark this answers is not in the document]
Comment (CharlieS1103) [reply docs/development/custom-apps.md:17]: No, this is just an overview of the components of a custom app

[the remark this answers is not in the document]
Comment (theRealPadster) [reply docs/development/custom-apps.md:17]: This is how custom apps are defined. It's where we got the idea for using one in Marketplace

File: docs/development/custom-apps.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
+---
+title: Custom Apps
+sidebar_position: 4
+---
+
+## How to start
+- Create an `index.js` and a `manifest.json`.
```
[the remark this answers is not in the document]
Comment (theRealPadster) [reply docs/development/custom-apps.md]: Added.

File: docs/development/custom-apps.md (hunk 2 above)
Comment (theRealPadster) [reply docs/development/custom-apps.md]: Updated.

**Review by afonsojramos — APPROVED** [review]

**Pull request merged**